### ISSUE TITLE
linter: `no-unassigned-vars` incorrectly reports Svelte `bind:this` targets

### ISSUE THREAD

**Issue by tisonkun** [oxc-project/oxc#26038]
### What version of Oxlint are you using?

1.79.0

### What command did you run?

`oxlint repro.svelte`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

No config file. This reproduces with the default correctness rules.

### What happened?

Given this Svelte component:

```svelte
<script lang="ts">
  let dialog: HTMLDialogElement;

  function openDialog() {
    dialog.showModal();
  }
</script>

<button type="button" onclick={openDialog}>Open</button>
<dialog bind:this={dialog}>Hello</dialog>
```

Oxlint reports:

```text
repro.svelte:2:7: warning eslint(no-unassigned-vars): 'dialog' is always 'undefined' because it's never assigned. help: Variable declared without assignment. Either assign a value or remove the declaration.
```

The expected result is no `no-unassigned-vars` diagnostic. Svelte assigns the element to `dialog` through `bind:this={dialog}` when the component is mounted, so the variable is not always `undefined`.

Oxlint currently lints only the `<script>` block and does not see the write in the template. The exact limitation and a config workaround were previously noted in https://github.com/oxc-project/oxc/issues/13017#issuecomment-3230976374, but that broader issue was closed as a duplicate of the language-support effort.

This seems analogous to:

- `no-unused-vars` skipping Svelte files because template uses are not visible: https://github.com/oxc-project/oxc/issues/17991
- `prefer-const` producing a false positive for `bind:this`: https://github.com/oxc-project/oxc/issues/19470, fixed by skipping Svelte and Vue files in https://github.com/oxc-project/oxc/pull/25148

Would it make sense for `no-unassigned-vars` to skip Svelte files until template writes can be analyzed?

### Comparison with the official Svelte ESLint parser

The same reproduction is handled correctly by ESLint when it uses the official Svelte parser.

Tested with:

- `eslint@10.9.0`
- `eslint-plugin-svelte@3.23.0`
- `svelte-eslint-parser@1.8.1`
- `typescript-eslint@8.67.0`

Minimal flat config:

```js
import { defineConfig } from 'eslint/config';
import svelte from 'eslint-plugin-svelte';
import ts from 'typescript-eslint';

export default defineConfig(
  svelte.configs.base,
  {
    files: ['**/*.svelte'],
    languageOptions: {
      parserOptions: {
        parser: ts.parser
      }
    },
    rules: {
      'no-unassigned-vars': 'error'
    }
  }
);
```

Running:

```console
$ eslint --no-config-lookup --config eslint.config.js repro.svelte
```

produces no diagnostics and exits with status 0. Running Oxlint on the same `repro.svelte` produces the warning shown above.

The official plugin uses `svelte-eslint-parser` to provide Svelte-aware AST and scope analysis:

- https://github.com/sveltejs/eslint-plugin-svelte
- https://github.com/sveltejs/svelte-eslint-parser

This suggests the diagnostic is specific to Oxlint's current script-only view of Svelte files rather than expected behavior of `no-unassigned-vars` when template semantics are available.

**Comment by tisonkun** [comment]
So the solution is effectively ignore svelte files for this rule?

Is there any block prevent us from implementing the rule correctly? Or there is no correct way to implement this.

**Comment by camc314** [comment]
> Is there any block prevent us from implementing the rule correctly? Or there is no correct way to implement this.

We don't understand any of the special svelte syntax for this file, so we can't understand that the variable is used in the template.

Please track https://github.com/oxc-project/oxc/discussions/21936
